osf/external/spam/tasks.py

```diff
@@ -8,25 +8,35 @@
 @celery_app.task()
 def reclassify_domain_references(notable_domain_id):
     from osf.models.notable_domain import DomainReference, NotableDomain
+    from osf.models.spam import SpamStatus
+    spammy_domains_set = set(NotableDomain.objects.filter(note=NotableDomain.Note.EXCLUDE_FROM_ACCOUNT_CREATION_AND_CONTENT).values_list('domain', flat=True))
     domain = NotableDomain.load(notable_domain_id)
     references = DomainReference.objects.filter(domain=domain)
     with transaction.atomic():
         for item in references:
+            item.is_triaged = domain.note != NotableDomain.Note.UNKNOWN
             if domain.note == NotableDomain.Note.EXCLUDE_FROM_ACCOUNT_CREATION_AND_CONTENT:
-                item.referrer.confirm_spam(save=True)
-                item.is_triaged = True
+                item.referrer.confirm_spam(save=False, domains=[domain.domain])
+            elif domain.note == NotableDomain.Note.UNKNOWN or domain.note == NotableDomain.Note.IGNORED:
+                if item.referrer.spam_status == SpamStatus.SPAM:
+                    item.referrer.spam_data['domains'].remove(domain.domain)
+                    if set(item.referrer.spam_data['domains']).isdisjoint(spammy_domains_set):
+                        item.referrer.unspam(save=False)
             item.save()
+            item.referrer.save()
 
 @celery_app.task()
 def check_resource_for_domains(guid, content):
     from osf.models import Guid, NotableDomain, DomainReference
     resource = Guid.load(guid).referent
     domains = {match.group('domain') for match in re.finditer(DOMAIN_REGEX, content)}
+    spammy_domains = []
     referrer_content_type = ContentType.objects.get_for_model(resource)
     mark_spam = False
     for domain in domains:
         domain, _ = NotableDomain.objects.get_or_create(domain=domain)
         if domain.note == NotableDomain.Note.EXCLUDE_FROM_ACCOUNT_CREATION_AND_CONTENT:
+            spammy_domains.append(domain.domain)
```

Comment:
nit: this makes `mark_spam` unnecessary, we can just check `if spammy_domains` below